util/testutil/workers/util.go

```diff
@@ -71,38 +90,12 @@ func runBuildkitd(ctx context.Context, conf *integration.BackendConfig, args []s
 	}
 	deferF.Append(stop)
 
-	if err := integration.WaitUnix(address, 15*time.Second, cmd); err != nil {
+	if err := integration.WaitSocket(address, 15*time.Second, cmd); err != nil {
 		return "", nil, err
 	}
 
-	deferF.Append(func() error {
-		f, err := os.Open("/proc/self/mountinfo")
-		if err != nil {
-			return errors.Wrap(err, "failed to open mountinfo")
-		}
-		defer f.Close()
-		s := bufio.NewScanner(f)
-		for s.Scan() {
-			if strings.Contains(s.Text(), tmpdir) {
-				return errors.Errorf("leaked mountpoint for %s", tmpdir)
-			}
-		}
-		return s.Err()
-	})
+	// separated out since it's not required in windows
+	mountInfo(deferF, tmpdir)
```

Comment:
The call to `deferF.Append()` should still be done in this function. The code we defer can be moved out, and we can define a noop on Windows.

So this should probably be:

```go
deferF.Append(func() error{
      return mountInfo(tmpdir)
    })
```

Where `mountInfo()` is separated out.

The idea is that it's much easier to understand that *this* function needs to defer a cleanup function (which is platform specific), than to call a function we must peek into to see that it will defer another function. It also feels a bit off passing the `MultiCloser{}` around.